### ISSUE TITLE
Change count direction

### ISSUE THREAD

**Issue by thomas-de-jager** [madhephaestus/ESP32Encoder#125]
Thank you for providing this library. I am not an advanced user, just started looking into this but it is promising. Using Lilygo t-display s3 with platformio and arduino library.

I was wondering if there is any way to change the pulse count direction? Boolean direction seems to be private, but I want to be able to change the counting direction.

As an example, change the behavior of the encoder if the encoder is rotating clockwise such that PCNT counts up (++) instead of down (--).

I was thinking of changing for example encoder.attachHalfQuad(16, 21); to encoder.attachHalfQuad(21, 16);, but this creates other issues like counter resetting and not detaching other encoder. A detach function would be an option too?

Basically I am looking to somehow change "enum pcnt_step_direction_t" from: https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/pcnt.html#_CPPv421pcnt_step_direction_t.

Looking forward to hearing from you. Thanks in advance!
Best regards,
Thomas

**Comment by thomas-de-jager** [comment]
Just noticed the encoder.detach() method exists but is not documented. Tested working.
